acceptance-tests/src/test/java/tech/pegasys/pantheon/tests/web3j/privacy/EthSignerAcceptanceTest.java

```diff
@@ -0,0 +1,87 @@
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.tests.web3j.privacy;
+
+import static tech.pegasys.pantheon.tests.acceptance.dsl.WaitUtils.waitFor;
+import static tech.pegasys.pantheon.tests.acceptance.dsl.transaction.eea.PrivateTransactionBuilder.EVENT_EMITTER_CONSTRUCTOR;
+import static tech.pegasys.pantheon.tests.web3j.privacy.PrivacyGroup.generatePrivacyGroup;
+
+import tech.pegasys.ethsigner.testutil.EthSignerTestHarness;
+import tech.pegasys.ethsigner.testutil.EthSignerTestHarnessFactory;
+import tech.pegasys.pantheon.ethereum.core.Address;
+import tech.pegasys.pantheon.ethsigner.EthSignerClient;
+import tech.pegasys.pantheon.tests.acceptance.dsl.privacy.PrivacyAcceptanceTestBase;
+import tech.pegasys.pantheon.tests.acceptance.dsl.privacy.PrivacyNet;
+import tech.pegasys.pantheon.util.bytes.BytesValue;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.Collections;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class EthSignerAcceptanceTest extends PrivacyAcceptanceTestBase {
+  private PrivacyNet privacyNet;
+
+  private EthSignerClient ethSignerClient;
+
+  @Before
+  public void setUp() throws Exception {
+    privacyNet =
+        PrivacyNet.builder(privacy, privacyPantheon, cluster, false).addMinerNode("Alice").build();
+    privacyNet.startPrivacyNet();
+
+    final EthSignerTestHarness ethsigner =
+        EthSignerTestHarnessFactory.create(
+            privacy.newFolder().toPath(),
+            "ethSignerKey--fe3b557e8fb62b89f4916b721be55ceb828dbd73.json",
+            privacyNet.getNode("Alice").getJsonRpcSocketPort().orElse(8545),
+            23606,
+            2018);
+    ethSignerClient = new EthSignerClient(ethsigner.getHttpListeningUrl());
+  }
+
+  @Test
+  public void privateSmartContractMustDeploy() throws IOException {
+    final BytesValue privacyGroupId = generatePrivacyGroup(privacyNet, "Alice");
+    final long nonce = privacyNet.getNode("Alice").nextNonce(privacyGroupId);
+
+    final String transactionHash =
+        ethSignerClient.eeaSendTransaction(
+            null,
+            BigInteger.valueOf(63992),
+            BigInteger.valueOf(1000),
+            EVENT_EMITTER_CONSTRUCTOR.toString(),
+            BigInteger.valueOf(nonce),
+            privacyNet.getEnclave("Alice").getPublicKeys().get(0),
+            Collections.emptyList(),
+            "restricted");
+
+    waitFor(
+        60,
+        () ->
+            privacyNet
+                .getNode("Alice")
+                .verify(eea.expectSuccessfulTransactionReceipt(transactionHash)));
+
+    final String expectedContractAddress =
+        Address.privateContractAddress(
+                privacyNet.getNode("Alice").getAddress(), nonce, privacyGroupId)
+            .toString();
+
+    privateTransactionVerifier
```

Comment:
Would be better if we can call a verify method on the node like we do for other conditions. So perhaps add a verify(final EeaCondition expected, final String transactionHash) method to PrivacyNode then 
 this could become something like privacyNetNode.verify(
        privateTransactionVerifier.validPrivateContractDeployed(expectedContractAddress),
        transactionHash);

Reply:
agreed - this work is being carried out in https://github.com/PegaSysEng/pantheon/pull/1864 I will not change it here

Reply:
i.e. once this is accepted I will rebase 1864 and change it accordingly